packages/ui/src/app/pages/WorkingGroups/MyRoles/MyRole.tsx

```diff
@@ -156,6 +162,11 @@ export const MyRole = () => {
                       Move Excess Tokens
                     </TransactionButton>
                   ))}
+                {isActive && shouldIncreaseStake && (
+                  <TransactionButton style="primary" size="small" onClick={onAddStakeClick}>
+                    Add Stake
+                  </TransactionButton>
+                )}
```

Comment:
Just a note on modal system. This might need a second thought probably later as we might not followed this rule strictly. If your modal is used in one place (I think that here is such case) you can just use `useToggle()` + render the modal right after the button `{showModal && <AddWorkerStakeModal />`.

Since, you already added this to global modals, there's no need to change it in this PR and I think that using global modal is OK for now.